### ISSUE TITLE
Inconsitent variable expansion for ${workspaceRoot}

### ISSUE THREAD

**Issue by tobse80** [microsoft/vscode-cmake-tools#1098]
### Brief Issue Summary

The way of how cmake-tools expands the variable `workspaceRoot` is inconsistent regarding path separators `\` vs. `/`. Especially when constructing custom variables based on `workspaceRoot` in e.g. _settings.json_, the observed behaviour is at least unexpected and can lead to further problems in e.g. a custom tool chain.

The behaviour seems to have changed from version 1.2.3 to 1.3.0.

### Expected:

1. Use cmake-tools version **1.2.3 or older**
2. Add these lines to _.vscode/settings.json_ in your project folder
```
   "cmake.buildDirectory": "${workspaceRoot}/Build",
   "cmake.installPrefix": "${workspaceRoot}/Install",
   "cmake.configureSettings": {
      "FOO_DIR": "${workspaceRoot}/Foo"
   }
```
3. Observe that all three directories use a `/` character as path separator consistently. This is the expected behaviour
```
CMAKE_CURRENT_BINARY_DIR: 'C:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Build'
CMAKE_INSTALL_PREFIX: 'c:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Install'
FOO_DIR: 'c:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Foo'
```

### Apparent Behavior:

1. Use cmake-tools version **1.3.0 or 1.3.1**
2. Use same lines in _.vscode/settings.json_ from above
3. Observe that `workspaceRoot` gets expanded with `\` when used inside `cmake.configureSettings`, while other contexts like `cmake.buildDirectory` or `cmake.installPrefix` the expected behaviour is still intact
```
CMAKE_CURRENT_BINARY_DIR: 'C:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Build'
CMAKE_INSTALL_PREFIX: 'c:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Install'
FOO_DIR: 'c:\Users\Tobse\Source\Repos\cmake-tools_dir-separator/Foo'
```

### CMake Tools Log

```
[main] Ordner wird konfiguriert: cmake-tools_dir-separator 
[driver] c:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Build/CMakeCache.txt wird entfernt
[driver] c:\Users\Tobse\Source\Repos\cmake-tools_dir-separator\Build\CMakeFiles wird entfernt
[proc] Befehl wird ausgeführt: C:\msys64\mingw64\bin\cmake.exe --no-warn-unused-cli -DFOO_DIR:STRING=c:\Users\Tobse\Source\Repos\cmake-tools_dir-separator/Foo -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_INSTALL_PREFIX:STRING=c:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Install -DCMAKE_C_COMPILER:FILEPATH=C:\msys64\mingw64\bin\gcc.exe -DCMAKE_CXX_COMPILER:FILEPATH=C:\msys64\mingw64\bin\g++.exe -Hc:/Users/Tobse/Source/Repos/cmake-tools_dir-separator -Bc:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Build -G Ninja
[cmake] Not searching for unused variables given on the command line.
[cmake] -- The CXX compiler identification is GNU 9.2.0
[cmake] -- Check for working CXX compiler: C:/msys64/mingw64/bin/g++.exe
[cmake] -- Check for working CXX compiler: C:/msys64/mingw64/bin/g++.exe -- works
[cmake] -- Detecting CXX compiler ABI info
[cmake] -- Detecting CXX compiler ABI info - done
[cmake] -- Detecting CXX compile features
[cmake] -- Detecting CXX compile features - done
[cmake] -- CMAKE_CURRENT_BINARY_DIR: 'C:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Build'
[cmake] -- CMAKE_INSTALL_PREFIX: 'c:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Install'
[cmake] -- FOO_DIR: 'c:\Users\Tobse\Source\Repos\cmake-tools_dir-separator/Foo'
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: C:/Users/Tobse/Source/Repos/cmake-tools_dir-separator/Build

```

### Platform and Versions

- **Operating System**: Windows 10 (18362.657)
- **CMake Version**: 3.16.4
- **VSCode Version**: 1.42.1 (system setup)
- **CMake Tools Extension Version**: 1.3.1
- **Compiler/Toolchain**: GCC 9.2.0, mingw64

**Comment by tobse80** [comment]
For reproducing the issue easily, you can clone a sample project here: https://github.com/tobse80/cmake-tools_dir-separator

**Comment by andreeis** [comment]
This should be fixed in CMake Tools 1.5.0 which was published today.
Let us know if you encounter any issues with this release.

**Comment by tobse80** [comment]
@andreeis, Thanks so far, CMake Tools 1.5.x work like a charm regarding this issue in former releases.
